src/main/java/net/mcreator/element/registry/ModElementRegistry.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.element.registry;
+
+import net.mcreator.element.types.GUI;
+import net.mcreator.element.types.*;
+import net.mcreator.ui.modgui.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static net.mcreator.element.registry.ModElementRegistry.BuiltInTypes.*;
+
+public class ModElementRegistry {

```

Comment:
Refactor to ModElementTypeRegistry, although it is a bit long, it describer the class more accurately

Reply:
Or maybe even better, keep this ModElementType and existing package, so eg. ModElementType.PROCEDURE like calls don't need refactoring. This will also reduce number of changed files a lot and make this reviewable. We can still refactor this later if it turns out it is needed